### PR TITLE
feat: paginate people

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Paginates the People Page to limit to 50 at a time

Some decisions I took:
- All the components are server side (including the navigation)
- Invalid page number throws them an error
- The pagination component only shows a " Previous | Next " in mobile screens
- The pagination component otherwise shows all the page numbers
- Added a new service to get the total People Count for page number renders

## Demo
I set the page size to 5 for easier testing!
https://www.loom.com/share/4874ce57a5254a62b5cdb8e540aca426?sid=c7f9b1c9-1132-43c2-a739-6ba28f899fa7

## Type of change
- [x] Enhancement (small improvements)

## Checklist
- [x] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/services/person.ts [hunk 1]
```diff
@@ -100,16 +101,20 @@ export const getPersonCached = async (personId: string) =>
     }
   )();
 
-export const getPeople = cache(async (environmentId: string): Promise<TPerson[]> => {
+export const getPeople = cache(async (environmentId: string, page: number = 1): Promise<TPerson[]> => {
   validateInputs([environmentId, ZId]);
   try {
+    const itemsPerPage = PERSONS_PER_PAGE;
     const personsPrisma = await prisma.person.findMany({
```
Comment (mattinannt) [inline packages/lib/services/person.ts:108]: please name `people` instead of `persons`

File: packages/lib/services/person.ts [hunk 2]
```diff
@@ -100,16 +101,20 @@ export const getPersonCached = async (personId: string) =>
...
       where: {
         environmentId: environmentId,
       },
       select: selectPerson,
+      take: itemsPerPage,
+      skip: itemsPerPage * (page - 1),
     });
-    if (!personsPrisma) {
+
+    if (!personsPrisma || personsPrisma.length === 0) {
```
Comment (mattinannt) [inline packages/lib/services/person.ts:117]: please remove this block and just return an empty array when no people are found

File: packages/lib/services/person.ts (hunk 1 above)
Comment (ShubhamPalriwala) [reply packages/lib/services/person.ts:108]: done!

File: packages/lib/services/person.ts (hunk 2 above)
Comment (ShubhamPalriwala) [reply packages/lib/services/person.ts:117]: done!

**Comment by ShubhamPalriwala** [comment]
If number of pages turn out to be more than 20, then

Before:
Screenshot_2023-09-19-16-00-02_5760x1080

After:
Screenshot_2023-09-19-16-02-39_5760x1080

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala thank you :-) I will merge the PR now!

**Pull request merged**